fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -456,6 +564,18 @@ public void removeExpiredTasks() {
         try {
             List<Task> currentTask = showTasks(null);
             for (Task task : currentTask) {
+                if (task.getType() == Constants.TaskType.PERIODICAL) {
+                    Schedule schedule = task.getSchedule();
+                    if (schedule == null) {
```

Comment:
why schedule could be null

Reply:
This code is just defensive programming, and it will not be null from a design point of view.

Reply:
if so, you could add Preconditions.checkState in ` task.getSchedule()` internal.

Reply:
when task type is PERIODICAL schedule can not be NULL, Preconditions.checkState will throw exception.
If this happens, it shouldn't throw an exception, but it should be able to delete successfully